api/provider/aws/service_read.go

```diff
@@ -1,48 +1,67 @@
 package aws
 
 import (
-	"fmt"
-
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
 	"github.com/quintilesims/layer0/common/models"
 )
 
 func (s *ServiceProvider) Read(serviceID string) (*models.Service, error) {
-	model := &models.Service{}
-
-	clusterName := addLayer0Prefix(s.Config.Instance(), model.EnvironmentID)
-	service, err := s.readService(clusterName, serviceID)
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
 	if err != nil {
 		return nil, err
 	}
 
-	model.DesiredCount = aws.Int64Value(service.DesiredCount)
-	model.RunningCount = aws.Int64Value(service.RunningCount)
-	model.PendingCount = aws.Int64Value(service.PendingCount)
+	fqEnvironmentID := addLayer0Prefix(s.Config.Instance(), environmentID)
+	clusterName := fqEnvironmentID
+	ecsService, err := s.readService(clusterName, serviceID)
+	if err != nil {
+		return nil, err
+	}
 
-	for _, deploy := range service.Deployments {
-		deployID := aws.StringValue(deploy.TaskDefinition)
-		//todo: convert deployid to layer0 deploy id
+	var deployments []models.Deployment
+	for _, d := range ecsService.Deployments {
+		taskDefinitionARN := aws.StringValue(d.TaskDefinition)
+		deployID, err := s.lookupDeployIDFromTaskDefinitionARN(taskDefinitionARN)
+		if err != nil {
+			return nil, err
+		}
 
-		deploy := models.Deployment{
-			DeploymentID: aws.StringValue(deploy.Id),
-			Created:      aws.TimeValue(deploy.CreatedAt),
-			Updated:      aws.TimeValue(deploy.UpdatedAt),
-			Status:       aws.StringValue(deploy.Status),
-			PendingCount: aws.Int64Value(deploy.PendingCount),
-			RunningCount: aws.Int64Value(deploy.RunningCount),
-			DesiredCount: aws.Int64Value(deploy.DesiredCount),
-			DeployID:     deployID,
+		deployment, err := s.newDeploymentModel(deployID)
+		if err != nil {
+			return nil, err
 		}
 
-		model.Deployments = append(model.Deployments, deploy)
+		deployment.Created = aws.TimeValue(d.CreatedAt)
+		deployment.DesiredCount = int(aws.Int64Value(d.DesiredCount))
+		deployment.PendingCount = int(aws.Int64Value(d.PendingCount))
+		deployment.RunningCount = int(aws.Int64Value(d.RunningCount))
+		deployment.Status = aws.StringValue(d.Status)
+		deployment.Updated = aws.TimeValue(d.UpdatedAt)
+
+		deployments = append(deployments, *deployment)
 	}
 
-	if err := s.updateWithTagInfo(model, serviceID); err != nil {
+	var loadBalancerID string
+	if len(ecsService.LoadBalancers) != 0 {
+		loadBalancer := ecsService.LoadBalancers[0]
+		loadBalancerName := aws.StringValue(loadBalancer.LoadBalancerName)
+		fqLoadBalancerID := loadBalancerName
+		loadBalancerID = delLayer0Prefix(s.Config.Instance(), fqLoadBalancerID)
+	}
+
+	model, err := s.newServiceModel(environmentID, loadBalancerID, serviceID)
```

Comment:
I made a similar comment in the WIP PR for this pattern, but I'll ask here as well: do the other CRUD operations _not_ need to call `newServiceModel`? If they do, we need to change the name for the functions in this pattern or generalize them more. Also not convinced that putting them directly in the various CRUD files makes sense versus (in this case) `service.go`.

Reply:
`newServiceModel()` should only be relevant for Read, just like the prior `populateModelTags()` was only relevant for Read. List, then, would replace `populateSummariesTags()` with something like `newServiceSummary()`, and that would likewise only be relevant to List. Since other operation functions don't call these functions, I don't think they belong in `service.go`, but should stay in their respective `service_<operation>.go` files.